src/Testcontainers.LocalStack/LocalStackBuilder.cs

```diff
@@ -0,0 +1,134 @@
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Testcontainers.Minio;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class LocalStackBuilder : ContainerBuilder<LocalStackBuilder, LocalStackContainer, LocalStackConfiguration>
+{
+    public const ushort LocalStackPort = 4566;
+    public const string LocalStackImage = "localstack/localstack:1.3.1";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LocalStackBuilder" /> class.
+    /// </summary>
+    public LocalStackBuilder()
+        : this(new LocalStackConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LocalStackBuilder" /> class.
+    /// </summary>
+    /// <param name="dockerResourceConfiguration">The Docker resource configuration.</param>
+    private LocalStackBuilder(LocalStackConfiguration dockerResourceConfiguration)
+        : base(dockerResourceConfiguration)
+    {
+        DockerResourceConfiguration = dockerResourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override LocalStackConfiguration DockerResourceConfiguration { get; }
+    
+    /// <summary>
+    /// Sets the LocalStack Default Ports Start.
+    /// </summary>
+    /// <param name="port">The LocalStack Default Ports Start.</param>
+    /// <returns>A configured instance of <see cref="LocalStackBuilder" />.</returns>
+    public LocalStackBuilder WithExternalServicePortStart(string port)
+    {
+        return Merge(DockerResourceConfiguration, new LocalStackConfiguration(externalServicePortStart: port))
+            .WithEnvironment("EXTERNAL_SERVICE_PORTS_START", port);
+    }
+    
+    /// <summary>
+    /// Sets the LocalStack Default Ports End.
+    /// </summary>
+    /// <param name="port">The LocalStack Default Ports End.</param>
+    /// <returns>A configured instance of <see cref="LocalStackBuilder" />.</returns>
+    public LocalStackBuilder WithExternalServicePortEnd(string port)
+    {
+        return Merge(DockerResourceConfiguration, new LocalStackConfiguration(externalServicePortEnd: port))
+            .WithEnvironment("EXTERNAL_SERVICE_PORTS_END", port);
+    }
+    
+    /// <summary>
+    /// Sets the LocalStack Services.
+    /// </summary>
+    /// <param name="services">The LocalStack services.</param>
+    /// <returns>A configured instance of <see cref="LocalStackBuilder" />.</returns>
+    public LocalStackBuilder WithServices(params AwsService[] services)
+    {
+        if (services?.Length is null)
+        {
+            return this;
+        }
+
+        string servicesNames = string.Join(",", services.Select(service => service.Name));
+        return Merge(DockerResourceConfiguration, new LocalStackConfiguration(services: services))
+            .WithEnvironment("SERVICES", servicesNames);
```

Comment:
```suggestion
        var localStackBuilder = Merge(DockerResourceConfiguration, new LocalStackConfiguration(services: services));
        return localStackBuilder.WithEnvironment("SERVICES", string.Join(",", localStackBuilder.DockerResourceConfiguration.Services.Select(service => service.Name)));
```

This one is a bit tricky. Your `SERVICES` environment variable and `DockerResourceConfiguration. Services` property diverges. The environment variable will contain the services from the `services` argument, but the property will contain all configured services. Imagine someone configures:

```csharp
_ = new LocalStackBuilder()
    .WithServices(AwsService.S3)
    .WithServices(AwsService.Route53)
    .Build();
```

Either the environment variable contains all services too, or we set the property to the latest configuration.

I removed the `null` check. IMO it should fail fast if someone passes `null` to the member. WDYT about the suggestion?